lib/_http_common.js

```diff
@@ -74,7 +74,11 @@ function parserOnHeadersComplete(versionMajor, versionMinor, headers, method,
     parser._url = '';
   }
 
-  parser.incoming = new IncomingMessage(parser.socket);
+  // Parser is also used by http client
+  var ParserIncomingMessage = parser.socket.server ?
```

Comment:
This a super minor nit but ideally we would check `parser.socket.server === null`.

Reply:
I checked it and it can be `undefined`, for example:

```
TypeError: Cannot read property 'IncomingMessage' of undefined
    at HTTPParser.parserOnHeadersComplete (_http_common.js:79:26)
    at TLSSocket.socketOnData (_http_client.js:437:20)
    at emitOne (events.js:115:13)
    at TLSSocket.emit (events.js:210:7)
    at addChunk (_stream_readable.js:264:12)
    at readableAddChunk (_stream_readable.js:251:11)
    at TLSSocket.Readable.push (_stream_readable.js:209:10)
    at TLSWrap.onread (net.js:591:20)
Command: out/Release/node /Users/pmarton/Documents/dev/node/test/parallel/test-async-wrap-GH13045.js
```

Reply:
Thanks for checking!